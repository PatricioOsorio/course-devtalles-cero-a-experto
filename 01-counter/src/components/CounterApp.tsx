import { useState } from 'react';

interface ICounterAppProps {
  initialValue: number;
}

export const CounterApp: React.FC<ICounterAppProps> = ({ initialValue }) => {
  const [value, setValue] = useState(initialValue);

  const handleAdd = () => setValue(value + 1);
  const handleSubstract = () => setValue(value - 1);
  const handleReset = () => setValue(initialValue);

  return (
    <article>
      <h1>CounterApp</h1>
      <h2>{value}</h2>
      <button aria-label='btn-add' onClick={handleAdd}>+1</button>
      <button aria-label='btn-substract' onClick={handleSubstract}>-1</button>
      <button aria-label='btn-reset' onClick={handleReset}>Reset</button>
    </article>
  );
};
