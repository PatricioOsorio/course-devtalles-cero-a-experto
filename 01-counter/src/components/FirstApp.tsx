import React from 'react';

interface IFirstAppProps {
  title: string;
  subtitle: string;
  name: string;
}

export const FirstApp: React.FC<IFirstAppProps> = ({ title, subtitle, name }) => {
  return (
    <div>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subtitle}</h2>
      <p>{name}</p>
    </div>
  );
};
