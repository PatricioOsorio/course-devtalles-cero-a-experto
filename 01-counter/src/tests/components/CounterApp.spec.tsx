import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { CounterApp } from '../../components/CounterApp';

describe('CounterApp', () => {
  const initialValue = 0;

  it('should match with the snapshot', () => {
    const { container } = render(<CounterApp initialValue={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  it('should show the initial value with 100', () => {
    render(<CounterApp initialValue={100} />);
    expect(screen.getByText('100')).toBeInTheDocument();
  });

  it('should increment with add button', () => {
    render(<CounterApp initialValue={initialValue} />);

    fireEvent.click(screen.getByRole('button', { name: 'btn-add' }));

    expect(screen.getByText('1')).toBeInTheDocument();
  });

  it('should decrement with substract button', () => {
    render(<CounterApp initialValue={initialValue} />);

    fireEvent.click(screen.getByRole('button', { name: 'btn-substract' }));

    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent('-1');
  });

  it('should reset with reset button', () => {
    render(<CounterApp initialValue={initialValue} />);

    fireEvent.click(screen.getByRole('button', { name: 'btn-add' }));
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(initialValue.toString());
  });
});
