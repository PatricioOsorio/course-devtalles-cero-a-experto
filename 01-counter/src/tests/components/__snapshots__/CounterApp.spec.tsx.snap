// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp should match with the snapshot 1`] = `
<div>
  <article>
    <h1>
      CounterApp
    </h1>
    <h2>
      0
    </h2>
    <button
      aria-label="btn-add"
    >
      +1
    </button>
    <button
      aria-label="btn-substract"
    >
      -1
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
  </article>
</div>
`;
